src/components/structures/MessagePanel.js

```diff
@@ -296,6 +296,12 @@ module.exports = React.createClass({
             // Wrap consecutive member events in a ListSummary
             if (isMembershipChange(mxEv)) {
                 let ts1 = mxEv.getTs();
+                // Ensure that the key of the MemberEventListSummary does not change with new
+                // member events. This will prevent it from being re-created necessarily, and
+                // instead will allow new props to be provided. In turn, the shouldComponentUpdate
+                // method on MELS can be used to prevent unnecessary renderings.
+                // `prevEvent` at this point is a non-member event or null.
+                const key = "memberlistsummary-" + (prevEvent ? prevEvent.getId() : "");
```

Comment:
If you're back-paginating, prevEvent will be null, so you use a fixed string: you suggested `memberlistsummary-`; I am suggesting `memberlistsummary-initial`.

Once you've stopped back-paginating (or rather, back-paginated past the group of member events), prevEvent is no longer null. With your suggestion, you change the key to `memberlistsummary-{prevEvent.getId()}`. I am suggesting instead using `memberlistsummary-{mxEv.getId()}`. It still only changes once: at the point you stop back-paginating.

Am I missing something?